From a33c04cfebb8fcc8b8dba1cbe464abca134176e9 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 9 Aug 2017 19:11:01 +0800
Subject: [PATCH 2378/5242] MLK-16197-3 video: fbdev: dcss: add fifo commit in
 dcss_set_par().

commit  255f016e9c8cc4af7d8237ed4fb3da3d64f8b19b from
https://source.codeaurora.org/external/imx/linux-imx.git

In dcss_set_par(), it will config the parameters related
with DEC400D/DTRC, DPR, SCALER, HDR10 and etc. So commit
all the registers configuration at the end of this function
to avoid mixing with later configurations which may cause
duplicating settings in one commit.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 7e7e81d..184f3c3 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2701,6 +2701,14 @@ static int dcss_set_par(struct fb_info *fbi)
 		}
 	}
 
+#if USE_CTXLD
+	ret = commit_to_fifo(fb_node, info);
+	if (ret) {
+		dev_err(&pdev->dev, "commit config failed\n");
+		goto out;
+	}
+#endif
+
 	if (saved_blank == FB_BLANK_UNBLANK) {
 		/* unblank the fb if need */
 		ret = dcss_blank(FB_BLANK_UNBLANK, fbi);
-- 
1.7.9.5

